vlog4j-graal/src/main/java/org/semanticweb/vlog4j/graal/GraalToVLog4JModelConverter.java

```diff
@@ -0,0 +1,214 @@
+	 * {@link Atom VLog4J Atom}.
+	 *
+	 * @param atom A {@link fr.lirmm.graphik.graal.api.core.Atom Graal Atom}
+	 * @return A {@link Atom VLog4J Atom}
+	 */
+	public static Atom convertAtom(final fr.lirmm.graphik.graal.api.core.Atom atom) {
+		final Predicate predicate = convertPredicate(atom.getPredicate());
+		final List<Term> terms = convertTerms(atom.getTerms());
+		return makeAtom(predicate, terms);
+	}
+
+	/**
+	 * Converts a {@link List} of {@link fr.lirmm.graphik.graal.api.core.Atom Graal
+	 * Atoms} into a {@link List} of {@link Atom VLog4J Atoms}.
+	 *
+	 * @param atoms A {@link List} of {@link fr.lirmm.graphik.graal.api.core.Atom
+	 *              Graal Atoms}.
+	 * @return A {@link List} of {@link Atom VLog4J Atoms}.
+	 */
+	public static List<Atom> convertAtoms(final List<fr.lirmm.graphik.graal.api.core.Atom> atoms) {
+		final List<Atom> result = new ArrayList<>();
+		for (final fr.lirmm.graphik.graal.api.core.Atom atom : atoms) {
+			result.add(convertAtom(atom));
+		}
+		return result;
+	}
+
+	/**
+	 * Converts a {@link AtomSet Graal AtomSet} into a {@link Conjunction VLog4J
+	 * Conjunction}.
+	 *
+	 * @param atomSet A {@link AtomSet Graal Atomset}
+	 * @return A {@link Conjunction VLog4J Conjunction}
+	 */
+	private static Conjunction convertAtomSet(final AtomSet atomSet) {
+		final List<Atom> result = new ArrayList<>();
+		try (CloseableIterator<fr.lirmm.graphik.graal.api.core.Atom> iterator = atomSet.iterator()) {
+			while (iterator.hasNext()) {
+				result.add(convertAtom(iterator.next()));
+			}
+		} catch (final IteratorException e) {
+			throw new GraalConvertException(
+					MessageFormat.format("Unexpected Iterator Exception when converting AtomSet {0}}.", atomSet), e);
+		}
+		return makeConjunction(result);
+	}
+
+	/**
+	 * Converts a {@link fr.lirmm.graphik.graal.api.core.Predicate Graal Predicate}
+	 * into a {@link Predicate VLog4J Predicate}.
+	 *
+	 * @param predicate A {@link fr.lirmm.graphik.graal.api.core.Predicate Graal
+	 *                  Predicate}
+	 * @return A {@link Predicate VLog4J Predicate}
+	 */
+	private static Predicate convertPredicate(final fr.lirmm.graphik.graal.api.core.Predicate predicate) {
+		return makePredicate(predicate.getIdentifier().toString(), predicate.getArity());
+	}
+
+	/**
+	 * Converts a {@link ConjunctiveQuery Graal Query} into a
+	 * {@link ConjunctiveGraalQueryToRule}. Answering a {@link ConjunctiveQuery
+	 * GraalConjunctiveQuery} is equivalent to adding a {@link Rule} with the query
+	 * atoms as the body and a single atom with a new predicate containing all the
+	 * query variables as the head. This rule head can then be used as a query atom
+	 * to obtain the results of the query.
+	 *
+	 * <p>
+	 * <b>WARNING</b>: The supplied {@code ruleHeadPredicateName} will be used to
+	 * create a predicate containing all answer variables from the {@code query}. If
+	 * you use this name in another predicate, you will get conflicts.
```

Comment:
"If you use this name in another predicate, you will get conflicts." sounds informal and confusing. If the same Predicate (predicate name and arity) is used within other rule heads of a knowledge base, the obtained query result may not be the expected Graal Conjunctive Query one.

Reply:
Done.